### ISSUE TITLE
Proliferation of pre-pull pods

### ISSUE THREAD

**Issue by arokem** [jupyterhub/zero-to-jupyterhub-k8s#814]
In a GKE deployment that has run for a few days, with ~70 users, we are now no longer able to log into some pods. Simultaneously, we have seen a proliferation of pre-pull pods. At this point, we have more than 50 of these. There's only 5 nodes in the cluster, with 8 cores per node. In case it matters, this cluster is set to autoscale. 

Moved from a conversation started in gitter. As pointed out there, might be related to https://github.com/jupyterhub/jupyterhub/issues/2043, with the exception that I am seeing 500 errors, not 503s

**Comment by consideRatio** [comment]
Regarding pre puller pods see https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/638 !

When you are seeing the 500 errors, what does it say in the logs?

**Comment by arokem** [comment]
When I run the command in #815 (is that what you meant?): 

`kubectl delete pods --namespace <NAMESPACE> --selector app=jupyterhub,component=image-puller`

They all just come back within minutes (seconds?)

**Comment by consideRatio** [comment]
Woops mistake, it should not be pods, it should be ds or daemonsets

**Comment by consideRatio** [comment]
@arokem if you want help with the 500 error, provide logs from the hub

```
# hub logs
kubectl logs -n yournamespace hub-....

# get logs if there are any of the singleuser server failing to startup
kubectl logs jupyter-...

# see whats described about it, the event section tends to
# have messages like "failed to pull image" and similar errors 
# if it never is able to startup
kubectl describe pod jupyter-...
```

**Comment by consideRatio** [comment]
```sh
# The script will delete resources that were meant to be temporary
# The bug that caused this is fixed in version 0.7b1 of the Helm chart
NAMESPACE=<YOUR-NAMESPACE>

resource_types="daemonset,serviceaccount,clusterrole,clusterrolebinding,job"
for bad_resource in $(kubectl get $resource_types --namespace $NAMESPACE | grep '/pre-pull' | awk '{print $1}');
do
    kubectl delete $bad_resource --namespace $NAMESPACE --now
done

kubectl delete $resource_types --selector hub.jupyter.org/deletable=true --namespace $NAMESPACE --now
```

**Comment by choldgraf** [comment]
woot, thanks @consideRatio!!!

**Comment by consideRatio** [comment]
This is fully resolved in 0.7.0-beta.1 :)
